From 46c59a068ba025f71e653d165d56e493fd26cc36 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 26 Aug 2014 12:49:40 -0400
Subject: [tools] cpupower: Remove redundant error check

Message-id: <1409057382-26432-30-git-send-email-prarit@redhat.com>
Patchwork-id: 89092
O-Subject: [RHEL7.1 PATCH BZ 1133893 29/31] cpupower: Remove redundant error check
Bugzilla: 1133893
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1133893

commit 059802f961db9717412b6958111ca1cd1865726e
Author: Peter Senna Tschudin <peter.senna@gmail.com>
Date:   Tue Jul 29 18:12:20 2014 +0200

    cpupower: Remove redundant error check

    Remove double checks, and move the call to print_error to the
    first check. Replace break by return, and return 0 on success.
    The simplified version of the coccinelle semantic patch that
    fixes this issue is as follows:

    // <smpl>
    @@
    expression E; identifier pr; expression list es;
    @@
    for(...;...;...){
    ...
    - if (E) break;
    + if (E){
    +  pr(es);
    +  break;
    + }
    ...
    }
    - if(E) pr(es);
    // </smpl>

    Signed-off-by: Peter Senna Tschudin <peter.senna@gmail.com>
    Signed-off-by: Thomas Renninger <trenn@suse.de>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/power/cpupower/utils/cpufreq-set.c b/tools/power/cpupower/utils/cpufreq-set.c
index a416de8..f656e58 100644
--- a/tools/power/cpupower/utils/cpufreq-set.c
+++ b/tools/power/cpupower/utils/cpufreq-set.c
@@ -320,12 +320,11 @@ int cmd_freq_set(int argc, char **argv)
 
   printf(_("Setting cpu: %d\n"), cpu);
   ret = do_one_cpu(cpu, &new_pol, freq, policychange);
-  if (ret)
-   break;
+  if (ret) {
+   print_error();
+   return ret;
+  }
  }
 
- if (ret)
-  print_error();
-
- return ret;
+ return 0;
 }
-- 
1.7.1